aea/protocols/base.py

```diff
@@ -405,3 +407,43 @@ def from_config(cls, configuration: ProtocolConfig, **kwargs) -> "Protocol":
         message_class.serializer = serialize_class
 
         return Protocol(configuration, message_class, **kwargs)
+
+
+class ProtocolSpecificationsRegistry:
+    """Registry to store protocol id and corresponding specification ids."""
+
+    PROTOCOL_TO_SPECIFICATION: Dict[PublicId, PublicId] = {}
+    SPECIFICATION_TO_PROTOCOL: Dict[PublicId, PublicId] = {}
```

Comment:
Note, this will potentially be shared with many AEAs in the same thread...

Reply:
fixed in https://github.com/fetchai/agents-aea/pull/2201